lib/active_fedora/associations/builder/association.rb

```diff
@@ -10,7 +10,7 @@ class << self
     # configure_dependency
     def self.build(model, name, options, &block)
       if model.dangerous_attribute_method?(name)
-        Deprecation.warn(ActiveFedora::Base, "You tried to define an association named #{name} on the model #{model.name}, but " \
+        ActiveFedora::Base.logger.error("You tried to define an association named #{name} on the model #{model.name}, but " \
```

Comment:
Yes, Rails raises an exception:

https://github.com/rails/rails/blob/master/activerecord/lib/active_record/associations/builder/association.rb#L22

We must have downgraded it to a deprecation (and I should have noted that in a comment..)